qa/tasks/cephfs/test_volumes.py

```diff
@@ -7653,6 +7701,818 @@ def test_subvolume_snapshot_clone_with_no_wait_not_enabled(self):
+
+        For this test, number of clones launched is LESS than number of async
+        thread.
+        '''
+        v = self.volname
+        sv = 'sv1'
+        ss = 'ss1'
+        c = self._gen_subvol_clone_name(3)
+
+        # make sure clone doesn't begin and cancellation happens when it is
+        # in pending state.
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_delay', 10)
+        # so that clone proceeds slowly even if somehow it begins, so that
+        # less, CPU is consumed.
+        self.config_set('mgr', 'mgr/volumes/clone_delay_every_regfile', 120)
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        sv_path = self.get_ceph_cmd_stdout(f'fs subvolume getpath {v} {sv}')
+        sv_path = sv_path[1:]
+        self._do_subvolume_io(sv, None, None, 3, 1024)
+        self.run_ceph_cmd(f'fs subvolume snapshot create {v} {sv} {ss}')
+        self._wait_till_rbytes_is_right(v, sv)
+
+        for i in c:
+            self.run_ceph_cmd(f'fs subvolume snapshot clone {v} {sv} {ss} {i}')
+        self._wait_for_clone_to_be_in_pending_state(c[2])
+
+        self._cancel_clones_and_assert(c)
+        # ensure that progress bar was removed from "ceph status" output
+        o = self.get_ceph_cmd_stdout('status --format json-pretty')
+        o = json.loads(o)
+        pevs = o['progress_events']
+        pevs = self._rm_global_recovery_event(pevs)
+        try:
+            self.assertEqual(len(pevs), 0)
+        except AssertionError:
+            log.info(f'pevs - {pevs}')
+            raise
+
+        try:
+            sv_path = sv_path.replace(sv, c[0])
+            o = self.mount_a.run_shell(f'ls -lh {sv_path}')
+            o = o.stdout.getvalue().strip()
+            log.info(o)
+            # ensure that all files were not copied
+            self.assertLess(len(o.split('\n')), 4)
+        except CommandFailedError as cfe:
+            # if command failed due to errno 2 (no such file or dir), that
+            # too is fine
+            if cfe.exitstatus == 2:
+                pass
+            else:
+                raise
+
+    def test_when_7_clones_are_cancelled(self):
+        '''
+        Test that both progress bars, one for ongoing clones and other for
+        ongoing+pending clones, are removed from the output of "ceph status"
+        command when clones are cancelled.
+
+        For this test, number of clones launched is MORE than number of async
+        thread.
+        '''
+        v = self.volname
+        sv = 'sv1'
+        ss = 'ss1'
+        c = self._gen_subvol_clone_name(7)
+
+        # make sure clone doesn't begin and cancellation happens when it is
+        # in pending state.
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_delay', 30)
+        # so that clone proceeds slowly even if somehow it begins, so that
+        # less, CPU is consumed.
+        self.config_set('mgr', 'mgr/volumes/clone_delay_every_regfile', 120)
```

Comment:
I don't see the implementation of `clone_delay_every_regfile` config option. I hope it is still a TODO.